### PR TITLE
Integrate reporting in Ghostery (Manifest V3)

### PR DESCRIPTION
**Pull request by philipp-classen**
Changes in reporting and anonymous-communication to integrate it in the Ghostery Manifest V3 Extension:
* Move "alive" message and country detection into reporting
* Move the clock to anonymous-communication and make it part of its public interface
* Bug fixes (e.g. problems with caching in the pattern-updater)

refs https://github.com/ghostery/ghostery-extension/pull/934

### CODE REVIEW

File: packages/reporting/src/country-provider.js [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    };
+
+    // Wrap browser APIs (primarily to swap them out in the unit tests):
+    // this._fetch works like the native "fetch".
+    //
+    // Notes:
+    // * will not work in NodeJs < 18, since trying to access "fetch" will throw
+    // * the extra argument wrapping was needed on Android (when simplify test
+    //   whether it is still needed in the Ghostery Android Browser)
+    this._fetch = _fetchImpl || ((...args) => fetch(...args));
```
Comment (chrmod) [inline packages/reporting/src/country-provider.js:68]:
```suggestion
    this._fetch = _fetchImpl || fetch;
```

File: packages/reporting/test/country-provider.spec.js [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+      statusText: 'Stub server has been configured to fail (this is expected).',
+    }));
+  }
+
+  function newCountryProvider() {
+    return new CountryProvider({
+      config,
+      storage,
+      storageKey,
+      _fetchImpl: fetchMock,
```
Comment (chrmod) [inline packages/reporting/test/country-provider.spec.js:70]: cannot we use something link this https://github.com/whotracksme/webextension-packages/blob/main/packages/anonymous-communication/test/server-public-key-accessor.spec.js#L79 instead?

**Review by smalluban — APPROVED** [review]
Generally, I don't see apparent mistakes in the code. However, I am unfamiliar with the logic behind the feature, so I cannot review the logic process.

File: packages/reporting/test/country-provider.spec.js (hunk 2 above)
Comment (philipp-classen) [reply packages/reporting/test/country-provider.spec.js:70]: OK, I'll update the PR to remove the dependency injection by mocking.

**Review by chrmod — APPROVED** [review]

**Pull request merged**